syntax = "proto3";

package Game;

service ExchangeGameState {
    rpc Exchange (GameState) returns (Action) {}
}

message GameState {
    enum ObjectDirection {
        UP = 0;
        RIGHT = 1;
        DOWN = 2;
        LEFT = 3;
    }
    
    bool isClosestObstacleBox = 1;
    int32 coinsNeeded = 2;
    ObjectDirection closestObstacle = 3;
    ObjectDirection closestCoin = 4;
    ObjectDirection closestEnemy = 5;
    ObjectDirection finishDirection = 6;
}

message Action {
    enum ActionDirection {
        UP = 0;
        RIGHT = 1;
        DOWN = 2;
        LEFT = 3;
    }

    ActionDirection moveDirection = 1;
    ActionDirection shotDirection = 2;
}

